replica/table.cc

```diff
@@ -764,7 +764,14 @@ future<> table::maybe_wait_for_sstable_count_reduction() {
     trigger_compaction();
     using namespace std::chrono_literals;
     auto start = db_clock::now();
-    co_await _sstables_changed.wait([&num_runs_with_memtable_origin, threshold] { return num_runs_with_memtable_origin() <= threshold; });
+    try {
+        auto wait_duration = (1000ms * count) / threshold;
+        co_await _sstables_changed.wait(wait_duration, [&num_runs_with_memtable_origin, threshold] { return num_runs_with_memtable_origin() <= threshold; });
+    } catch (const condition_variable_timed_out&) {
```

Comment:
Doesn't this break the condition? If we're slow, we can still end up with an unbootable node now.

Reply:
Never mind, we can just rely on the bucket-based condition and checking that regular compaction is possible and wait until the condition is satisfied with no timeout.